### ISSUE TITLE
Slow intellisense after adding GraphQLSP

### ISSUE THREAD

**Issue by traed** [0no-co/GraphQLSP#254]
Intellisense has become extremely slow after adding GraphQLSP and gql.tada, up from almost instant feedback in VS Code to sometimes 10+ seconds. I'm not entirely sure how to troubleshoot this, but just reverting the commit where I added this returns the performance back to normal so I'm fairly certain the problem is not with some other VS Code extension or something like that.

Do you have any idea what might be causing this? 

Installed versions:
- @0no-co/graphqlsp: 1.4.1
- gql.tada: 1.2.1
- typescript: 5.3.3
- graphql: 16.8.1

My tsconfg is as follows:

```
// base.json
{
  "$schema": "https://json.schemastore.org/tsconfig",
  "display": "Default",
  "compilerOptions": {
    "composite": false,
    "declaration": true,
    "declarationMap": true,
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "allowImportingTsExtensions": true,
    "noEmit": true,
    "inlineSources": false,
    "isolatedModules": true,
    "module": "ESNext",
    "moduleResolution": "Bundler",
    "noUnusedLocals": false,
    "noUnusedParameters": false,
    "preserveWatchOutput": true,
    "skipLibCheck": true,
    "strict": true,
    "strictNullChecks": true
  },
  "exclude": ["node_modules"]
}

// next.json
{
  "$schema": "https://json.schemastore.org/tsconfig",
  "display": "Next.js",
  "extends": "./base.json",
  "compilerOptions": {
    "plugins": [
      {
        "name": "next"
      }
    ],
    "allowJs": true,
    "declaration": false,
    "declarationMap": false,
    "incremental": true,
    "jsx": "preserve",
    "lib": [
      "dom",
      "dom.iterable",
      "esnext"
    ],
    "module": "esnext",
    "resolveJsonModule": true,
    "strict": true,
    "strictFunctionTypes": false,
    "target": "es5"
  },
  "include": [
    "src",
    "next-env.d.ts"
  ],
  "exclude": [
    "node_modules"
  ]
}

{
  "extends": "./nextjs.json",
  "compilerOptions": {
    "plugins": [
      {
        "name": "next"
      },
      {
        "name": "@0no-co/graphqlsp",
        "schema": "http://localhost:8080/graphql",
        "tadaOutputLocation": "./src/graphql-env.d.ts",
        "template": "gql"
      }
    ]
  },
  "include": ["next-env.d.ts", "**/*.ts", "**/*.tsx", ".next/types/**/*.ts"],
  "exclude": ["node_modules"]
}
```

**Comment by JoviDeCroock** [comment]
Is it `tada` being slow or the LSP? If you turn off the LSP is sit still slow? Not sure what you mean with intellisense - just auto-complete or?

**Comment by JoviDeCroock** [comment]
You turn it off by removing it from the `tsconfig` plugins, then it just uses `tada`

**Comment by traed** [comment]
Ok, I removed it and reloaded the vs code window. That had no effect.

**Comment by JoviDeCroock** [comment]
All right, then the issue is with `tada`, would you mind making a minimal reproduction for us? We have a few assumptions at the moment and are going to validate them.

**Comment by kitten** [comment]
Related issue: https://github.com/0no-co/gql.tada/issues/89

**Comment by kitten** [comment]
@traed: Do you have specific reproduction steps here?
Intellisense and type inference seem pretty snappy to me in both `1.2.1` and `1.3.0` which we just released

**Comment by traed** [comment]
As I just said, it doesn't seem super slow in the example since it's a quite small project. But the same code in our larger project is much slower. I don't know how to provide a small sample but make it bit enough for the bad performance to be noticeable...


https://github.com/0no-co/GraphQLSP/assets/7236190/6731e7a5-839e-465d-8491-90bf7d713126

**Comment by kitten** [comment]
I can't seem to reproduce this exact issue you're showing in the video, since it should infer that `content` is non-null from it being in the `"creator"` case switch. Either way, in `1.2.1` inspecting the type of `content` (either directly or by adding `content;` on the line above) is a little slow. This does seem to be fixed however in `1.3.0`. Have you tried upgrading yet?

**Comment by kitten** [comment]
This does seem to now be faster given the latest version. For now, I haven't identified any potential speed ups that could help us here, and a lot of the ongoing problems with performance often come down to some expensive types being re-evaluated (rather than TS using its cache, afaict)
I'll close this for now. See #89 for more information
